src/index.js

```diff
@@ -20,12 +20,30 @@ const client = new ApolloClient({
   cache: new InMemoryCache(),
 });
 
-ReactDOM.render(
-  <ApolloProvider client={client}>
-    <App />
-  </ApolloProvider>,
-  document.getElementById('root')
-);
+function renderApp(config) {
+  ReactDOM.render(
+    <ApolloProvider client={client}>
+      <App {...config} />
+    </ApolloProvider>,
+    document.getElementById('root')
+  );
+}
+
+fetch('config.json')
+  .then(response => {
```

Comment:
Do we have a transpiler for `await` syntax in place? That might alleviate the added number of lines to read here, if `fetch()` returns an ES6 promise and not a jQuery one.

Reply:
Never mind, this suggestion is probably not useful.

Reply:
As we discussed, I thought the refactoring would look something like this:
```jsx
(async function () {
  let config = undefined
  try {
    let response = await fetch('config.json')
    if (!response.ok) {
      throw Error('Got ' + response.status + ' http status')
    }
    config = await response.json()
  } catch(err) {
    console.log(
      "Using default application configuration as config.json hasn't been found or parsed:",
      err
    )
  } 
  return config
})().then(config => ReactDOM.render(
  <ApolloProvider client={client}>
    <App {...config} />
  </ApolloProvider>,
  document.getElementById('root')
))
```

Note that (in addition to being an untested sketch) this doesn't catch any exceptions that `ReactDOM.render()` might throw.

Reply:
Factoring out the `await`s again:

```jsx
fetch('config.json')
  .then(function (response) {
    if (!response.ok) {
      throw Error('Got ' + response.status + ' http status')
    }
    return response.json()
  }).catch(err) {
    console.log(
      "Using default application configuration as config.json hasn't been found or parsed:",
      err
    )
    return Promise.resolve(undefined)
  }).then(config => ReactDOM.render(
    <ApolloProvider client={client}>
      <App {...config} />
    </ApolloProvider>,
    document.getElementById('root')
  ))
```

Reply:
I don't seem to have a resolve button, but I see that you followed up on this :heavy_check_mark: